### PR TITLE
Bonnie/reset password logic

### PR DESCRIPTION
**Pull request by bonnie-chin**
**Work done:**
- Got authentication working w/ the help of Megan :D
- Navigates to appropriate modals upon password reset success / failure
- Got rid of all usages of requirementsAreFulfilled

**There's some styling quirks with** 
- left margin of input field 
- padding of success/failure modals
- width of the login button upon a successful password reset (uses the common modal but I didn't wanna mess with that)
...not sure what the best move here is 👀

image
image
image

### CODE REVIEW

File: client/src/pages/ResetPasswordModal.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  const handleClick = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (requirements.length == 0) {
+      const URLParams = new URLSearchParams(window.location.href);
+      const actionCode = URLParams.get("oobCode");
+      console.log(actionCode);
+      if (actionCode) {
+        handlePasswordReset(actionCode, password)
+          .then((resp: boolean) => {
+            console.log(resp);
```
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: stray logs on 31 & 35

File: client/src/pages/ResetPasswordModal.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            </form>
+          </div>} />
+        <LogoModal show={passwordResetAttempted && passwordResetSuccessful} title={"Password Reset Successful"} subtitle={''} handleClose={() => setRedirect("/")} body={
+          <span>
+            <div className="text">
+              Your password has now been changed. You may now close this window or login with your new password using the button below.
+            </div>
+            <button className="button" onClick={() => setRedirect("/login")}>
+              Log in
+        </button>
```
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: Nit: indent this </button> in a couple times and the alignment will be :fire:

File: client/src/pages/ResetPasswordModal.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                role="link"
+                className="button signup"
+              >
+                Reset password
+              </button>
+              <div>
+                <div
+                  
+                  className="text redirect center"
+                >
```
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: Nit: let's put this all on one line

File: client/src/pages/ResetPasswordModal.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+import { TextField } from "../components/atoms/TextField";
+
+const ResetPasswordModal: FunctionComponent = () => {
+  const [password, setPassword] = useState(""); // can we put these all in an array somehow?????
+  const [hidePassword, setHidePassword] = useState(true);
+  const [hasOneLowerCase, setHasOneLowerCase] = useState(false);
+  const [hasOneUpperCase, setHasOneUpperCase] = useState(false);
+  const [hasOneNumber, setHasOneNumber] = useState(false);
+  const [hasOneSymbol, setHasOneSymbol] = useState(false);
+  const [hasTwelveCharacterMin, setHasTwelveCharacterMin] = useState(false);
```
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: @meganniu @bonnie-chin An idea I had to combine all of these test states into one is to encode them into a single state variable which contains a 5-digit binary number. Each digit will correspond to a test (using an object), modulo operators combined with addition and subtraction can be used to change the status of each test/determine which requirements to display. Probably not worth the work now but could be done in the future.

File: client/src/pages/ResetPasswordModal.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import React, { FunctionComponent, useState } from "react";
+import { Redirect } from "react-router-dom";
+
+import ConfirmationModal from "./ConfirmationModal";
```
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: Nit: Remove unused import

File: client/src/pages/ResetPasswordModal.tsx (hunk 4 above)
Comment (chamod-gamage) [inline client/src/pages/ResetPasswordModal.tsx]: Are we actually relying on these in any way? From what I can tell, the `forEach` loop inside `onChangePass` updates the state but ends up setting the requirements within the function without needed the individual states. To me, looks like everything works even if the following lines (inclusive) are deleted: 13-17, 60-66, 83. 

Also, ignore the other comment I made if this is true.

**Review by chamod-gamage — CHANGES_REQUESTED** [review]
Great work here! Just a few nitpicks and the individual states for password requirements issue (see the comment I made), which is causing CircleCI to be annoying rn. 
Also, great job with the function that reads the oobCode from the URL!
I'll also take a look at the styling sometime today too

File: client/src/pages/ResetPasswordModal.tsx (hunk 4 above)
Comment (chamod-gamage) [reply client/src/pages/ResetPasswordModal.tsx]: Ignore this if the comment at the bottom allows you to delete this lines completely.

Comment (meganniu) [reply client/src/pages/ResetPasswordModal.tsx]: Yup! ended up deleting those lines :)

File: client/src/pages/ResetPasswordModal.tsx [hunk 6]
```diff
@@ -3,54 +3,64 @@ import React, { FunctionComponent, useState } from "react";
...
-  // const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
+  const [password, setPassword] = useState(""); // can we put these all in an array somehow?????
   const [hidePassword, setHidePassword] = useState(true);
-  const [hasOneLowerCase, setHasOneLowerCase] = useState(false);
-  const [hasOneUpperCase, setHasOneUpperCase] = useState(false);
-  const [hasOneNumber, setHasOneNumber] = useState(false);
-  const [hasOneSymbol, setHasOneSymbol] = useState(false);
-  const [hasTwelveCharacterMin, setHasTwelveCharacterMin] = useState(false);
+  // const [hasOneLowerCase, setHasOneLowerCase] = useState(false);
```
Comment (meganniu) [inline client/src/pages/ResetPasswordModal.tsx]: Could we delete these comments?

File: client/src/pages/ResetPasswordModal.tsx [hunk 7]
```diff
@@ -3,54 +3,64 @@ import React, { FunctionComponent, useState } from "react";
...
   };
 
-  const requirementToStateSetterMap = new Map([
-    ['lowerCase', setHasOneLowerCase],
-    ['upperCase', setHasOneUpperCase],
-    ['number', setHasOneNumber],
-    ['symbol', setHasOneSymbol],
-    ['twelveCharacters', setHasTwelveCharacterMin],
-  ]);
+  // const requirementToStateSetterMap = new Map([
```
Comment (meganniu) [inline client/src/pages/ResetPasswordModal.tsx]: Also delete these

File: client/src/pages/SignInModal.tsx [hunk 8]
```diff
@@ -20,6 +20,7 @@ const SignInModal: FunctionComponent = () => {
   const handleClick = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     signIn(email, password).then((res) => {
+      console.log(res)
```
Comment (meganniu) [inline client/src/pages/SignInModal.tsx]: Remove console.log

File: client/src/services/auth.ts [hunk 9]
```diff
@@ -107,6 +120,8 @@ export const signIn = async (
   email: string,
   password: string
 ): Promise<{ email: string; password: string }> => {
+  console.log(email)
```
Comment (meganniu) [inline client/src/services/auth.ts]: Remove console.logs

**Review by meganniu — APPROVED** [review]
LGTM, thanks @bonnie-chin!!!
btw i resolved some merge conflicts for you. Can you pull this branch and make sure that password reset works before merging?

**Review by chamod-gamage — APPROVED** [review]
Great job - lgtm!

**Comment by bonnie-chin** [comment]
> LGTM, thanks @bonnie-chin!!!
> btw i resolved some merge conflicts for you. Can you pull this branch and make sure that password reset works before merging?

@meganniu uHhHHh slight problem - I don't know if it's me having to run the server again or smth but it doesn't work 😬

**Pull request merged**